build_modules/lib/src/module_library.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/analyzer.dart';
+import 'package:build/build.dart';
+import 'package:meta/meta.dart';
+
+class ModuleLibrary {
+  /// The AssetId of the original Dart source file.
+  final AssetId id;
+
+  /// Whether this library can be imported.
+  ///
+  /// This will be false if the source file is a "part of", or imports code that
+  /// can't be used outside the SDK.
```

Comment:
Curious, what would be an example import that can't be used outside the SDK? Are you referring to `dart:_` type imports?

Reply:
Yeah this is specifically referring to the checks at https://github.com/dart-lang/build/pull/1649/files#diff-08a717a90be39b7bdd847098ce7bc663R92